python-threatexchange/threatexchange/hashing/pdq_faiss_matcher.py

```diff
@@ -101,6 +101,52 @@ def search(
             for i in range(len(query_vectors))
         ]
 
+    def search_with_distance(
+        self,
+        queries: t.Sequence[PDQ_HASH_TYPE],
+        threshhold: int,
+        return_as_ids: bool = False,
+    ):
+        """
+        Same as search however instead of returning only sequence of matches per query
```

Comment:
(nit, ignorable): Recommended way of doing long docstrings is to have a one-line at the top that is a short summary, and then more details below: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings